pkg/server/server_presentation_test.go

```diff
@@ -55,16 +56,151 @@ func TestPresentationAPI(t *testing.T) {
 	assert.NoError(t, err)
 
 	for _, test := range testutil.TestDatabases {
-		t.Run(test.Name, func(t *testing.T) {
-			t.Run("Create, Get, and Delete PresentationDefinition", func(tt *testing.T) {
-				s := test.ServiceStorage(tt)
-				pRouter, _ := setupPresentationRouter(tt, s)
+		t.Run(test.Name, func(tt *testing.T) {
+
+			tt.Run("Verify a Verifiable Presentation", func(ttt *testing.T) {
+				db := test.ServiceStorage(ttt)
+				presRouter, _ := setupPresentationRouter(ttt, db)
+
+				// first, create a credential using the service
+				keyStoreService, _ := testKeyStoreService(ttt, db)
+				didService, _ := testDIDService(ttt, db, keyStoreService, nil)
+				schemaService := testSchemaService(ttt, db, keyStoreService, didService)
+				credRouter := testCredentialRouter(ttt, db, keyStoreService, didService, schemaService)
+
+				issuerDID, err := didService.CreateDIDByMethod(context.Background(), did.CreateDIDRequest{
+					Method:  didsdk.KeyMethod,
+					KeyType: crypto.Ed25519,
+				})
+				assert.NoError(ttt, err)
+				assert.NotEmpty(ttt, issuerDID)
+
+				// good request
+				createCredRequest := router.CreateCredentialRequest{
+					Issuer:               issuerDID.DID.ID,
+					VerificationMethodID: issuerDID.DID.VerificationMethod[0].ID,
+					Subject:              "did:car:911",
+					Data: map[string]any{
+						"firstName": "Frank",
+						"lastName":  "Ocean",
+					},
+					Expiry: time.Now().Add(24 * time.Hour).Format(time.RFC3339),
+				}
+				requestValue := newRequestValue(ttt, createCredRequest)
+				req := httptest.NewRequest(http.MethodPut, "https://ssi-service.com/v1/credentials", requestValue)
+				w := httptest.NewRecorder()
+				c := newRequestContext(w, req)
+				credRouter.CreateCredential(c)
+				assert.True(ttt, util.Is2xxResponse(w.Code))
+
+				var createResp router.CreateCredentialResponse
+				err = json.NewDecoder(w.Body).Decode(&createResp)
+				assert.NoError(ttt, err)
+
+				assert.NotEmpty(ttt, createResp.CredentialJWT)
+				assert.NoError(ttt, err)
+				assert.Equal(ttt, createResp.Credential.Issuer, issuerDID.DID.ID)
+
+				holderSigner, holderDID := getSigner(ttt)
+				testPresentation := credential.VerifiablePresentation{
+					Context: []string{"https://www.w3.org/2018/credentials/v1",
+						"https://w3id.org/security/suites/jws-2020/v1"},
+					Type:   []string{"VerifiablePresentation"},
+					Holder: holderDID.String(),
+				}
+
+				// invalid verifiable presentation with no credentials
```

Comment:
Here and elsewhere, better to use `t.Run("verifying verifiable presentation with no credentials returns false", func ...` if possible.

Reply:
was following the pattern you had below...